### PR TITLE
add multiple ad unit integration for interstitital and banner ads

### PR DESCRIPTION
**Pull request by guylis-ironsrc**

### CODE REVIEW

File: iOS/Objective C/IronSourceDemoApp/DemoViewController.m [hunk 1]
```diff
@@ -77,25 +87,33 @@ - (void)setupIronSourceSdk {
...
+    // Init the SDK when implementing the Multiple Ad Units Interstitial and Banner API, and Rewarded using legacy APIs
+    LPMInitRequestBuilder *requestBuilder = [[LPMInitRequestBuilder alloc] initWithAppKey:kAppKey];
+    [requestBuilder withLegacyAdFormats:@[IS_REWARDED_VIDEO]];
+
+    LPMInitRequest *initRequest = [requestBuilder build];
+    
+    [self logMethodName:[NSString stringWithFormat:@"init ironSource SDK with appKey: %@", kAppKey]];
+    [LevelPlay initWithRequest:initRequest completion:^(LPMConfiguration *_Nullable config, NSError *_Nullable error){
+    
+        if(error) {
```
[the remark this answers is not in the document]
Comment (guylis-ironsrc) [reply iOS/Objective C/IronSourceDemoApp/DemoViewController.m:106]: guard does not exist in objective c, only in swift

File: iOS/Objective C/IronSourceDemoApp/DemoBannerAdDelegate.m [hunk 2]
```diff
@@ -23,54 +23,71 @@ - (instancetype)initWithDelegate:(id<DemoViewControllerDelegate>)delegate {
  Called after each banner ad has been successfully loaded, either a manual load or banner refresh
  @param adInfo The info of the ad.
  */
-- (void)didLoad:(ISBannerView *)bannerView
-     withAdInfo:(ISAdInfo *)adInfo {
+- (void)didLoadAdWithAdInfo:(nonnull LPMAdInfo *)adInfo {
     logCallbackName(@"adInfo = %@", adInfo);
-    [self.delegate setAndBindBannerView:bannerView];
+    [self.delegate didLoadBannerAd];
     [self.delegate setEnablementForButton:LoadBannerButtonIdentifier
```
Comment (ArielBarsheshetIS) [inline iOS/Objective C/IronSourceDemoApp/DemoBannerAdDelegate.m:29]: We can remove it and to do its logic on the `didLoadBannerAd`

File: iOS/Objective C/IronSourceDemoApp/DemoViewController.m [hunk 3]
```diff
@@ -7,27 +7,32 @@
...
 
 @property (nonatomic, strong) DemoImpressionDataDelegate    *impressionDataDelegate;
 
 @property (nonatomic, strong) ISPlacementInfo           *rewardedVideoPlacementInfo;
-@property (nonatomic, strong) ISBannerView              *bannerView;
+
+@property (nonatomic, strong) LPMBannerAdView             *bannerAdView;
+@property (nonatomic, strong) LPMAdSize                   *bannerSize;
+@property (nonatomic, strong) LPMInterstitialAd           *interstitialAd;
+
```
Comment (ArielBarsheshetIS) [inline iOS/Objective C/IronSourceDemoApp/DemoViewController.m:35]: remove line

File: iOS/Objective C/IronSourceDemoApp/DemoViewController.m [hunk 4]
```diff
@@ -7,27 +7,32 @@
...
+@property (nonatomic, strong) DemoBannerAdDelegate            *bannerAdViewDelegate;
 
 @property (nonatomic, strong) DemoImpressionDataDelegate    *impressionDataDelegate;
 
 @property (nonatomic, strong) ISPlacementInfo           *rewardedVideoPlacementInfo;
-@property (nonatomic, strong) ISBannerView              *bannerView;
+
+@property (nonatomic, strong) LPMBannerAdView             *bannerAdView;
+@property (nonatomic, strong) LPMAdSize                   *bannerSize;
+@property (nonatomic, strong) LPMInterstitialAd           *interstitialAd;
```
Comment (ArielBarsheshetIS) [inline iOS/Objective C/IronSourceDemoApp/DemoViewController.m:34]: The properties are divided to sections so I would combine the banner properties and interstitial properties on different sections.

File: iOS/Objective C/IronSourceDemoApp/DemoViewController.m [hunk 5]
```diff
@@ -77,25 +87,33 @@ - (void)setupIronSourceSdk {
...
     
-    // To initialize specific ad units:
-    // [IronSource initWithAppKey:kAppKey adUnits:@[IS_REWARDED_VIDEO, IS_INTERSTITIAL, IS_BANNER] delegate:self.initializationDelegate];
+    // Init the SDK when implementing the Multiple Ad Units Interstitial and Banner API, and Rewarded using legacy APIs
+    LPMInitRequestBuilder *requestBuilder = [[LPMInitRequestBuilder alloc] initWithAppKey:kAppKey];
+    [requestBuilder withLegacyAdFormats:@[IS_REWARDED_VIDEO]];
+
+    LPMInitRequest *initRequest = [requestBuilder build];
+    
+    [self logMethodName:[NSString stringWithFormat:@"init ironSource SDK with appKey: %@", kAppKey]];
```
Comment (ArielBarsheshetIS) [inline iOS/Objective C/IronSourceDemoApp/DemoViewController.m:103]: I would move this line before the requestBuilder, and then the initRequest parameter will be close the [LevelPlay initWithRequest] without spaces.

File: iOS/Objective C/IronSourceDemoApp/DemoViewController.m [hunk 6]
```diff
@@ -114,62 +132,75 @@ - (IBAction)showRewardedVideoButtonTapped:(id)sender {
...
 }
 
+- (void) createInterstititalAd {
+    self.interstitialAd = [[LPMInterstitialAd alloc] initWithAdUnitId:kInterstitialAdUnitId];
+    self.interstitialAdDelegate = [[DemoInterstitialAdDelegate alloc] initWithDelegate:self];
+    self.interstitialAd.delegate = self.interstitialAdDelegate;
+    
+    [self setEnablementForButton:LoadInterstitialButtonIdentifier
+                                   enable:YES];
+
```
Comment (ArielBarsheshetIS) [inline iOS/Objective C/IronSourceDemoApp/DemoViewController.m:142]: remove

**Review by ArielBarsheshetIS — APPROVED** [review]

File: iOS/Objective C/IronSourceDemoApp/DemoRewardedVideoAdDelegate.m [hunk 7]
```diff
@@ -34,8 +34,8 @@ - (void)hasAvailableAdWithAdInfo:(ISAdInfo *)adInfo {
  */
 - (void)hasNoAvailableAd {
     logCallbackName();
-    [self.delegate setEnablementForButton:ShowRewardedVideoButtonIdentifier
-                                   enable:NO];
+     [self.delegate setEnablementForButton:ShowRewardedVideoButtonIdentifier
```
Comment (ArielBarsheshetIS) [inline iOS/Objective C/IronSourceDemoApp/DemoRewardedVideoAdDelegate.m:37]: looks like a redundant space

**Pull request merged**